syntax = "proto3";

package auth;

service AuthService {
  rpc RegisterUser(RegisterUserRequest) returns (RegisterUserResponse);
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc VerifyToken(VerifyTokenRequest) returns (VerifyTokenResponse);
}

message RegisterUserRequest {
   string name = 1;
   string email = 2;
   string password = 3;
}

message RegisterUserResponse {
   bool success = 1;
   string message = 2;
   int32 user_id = 3;
}

message LoginRequest {
   string email = 1;
   string password = 2;
}

message LoginResponse {
   string token = 1;
   bool admin = 2;
   bool success = 3;
   string message = 4;
}

message VerifyTokenRequest {
   string token = 1;
}

message VerifyTokenResponse {
   bool is_valid = 1;
   bool is_admin = 2;
   int32 user_id = 3;
}
